deps/oblib/src/lib/oblog/ob_log_compressor.cpp

```diff
@@ -231,6 +231,12 @@ void ObLogCompressor::log_compress()
             unlink(file_name->ptr());
           }
         }
+        if (!compression_file_name.empty()){
+          ob_free(compression_file_name.ptr());
```

Comment:
the key of this bug is: `get_compression_file_name` have allocated the memory, but it doesn't manage the life cycle of the allocated memory.

your fix only free the memory in this routine, but can not prevent the leaks caused by calling `get_compression_file_name` in other routines.

advice in my opinion: the caller of `get_compression_file_name` allocate and free the memory itself.  `get_compression_file_name` do not allocate the memory, it only checks whether the capacity of the input buffer is enough.